src/pages/Login/Login.jsx

```diff
@@ -0,0 +1,114 @@
+import "./Login.scss";
+import Navbar from "../../Components/Navbar/Navbar";
+import { useState } from "react";
+import { useAuth } from "../../context/AuthProvider";
+import { useNavigate, useLocation } from "react-router-dom";
+import loadingAnimation from "../../assets/loader.svg";
+import logo from "../../assets/prerogative-logo.svg";
+import logoText from "../../assets/dark-logo-with-text.svg";
+import { CgDanger } from "react-icons/cg";
+import { useForm } from "react-hook-form";
+
+function Login() {
+  const { loginWithCreds } = useAuth();
+  const [loading, setLoading] = useState(false);
+  const [authFaliureMessage, setAuthFaliureMessage] = useState(null);
+  const [userNotFoundMessage, setUserNotFoundMessage] = useState(null);
+  const navigate = useNavigate();
+  const { register, handleSubmit } = useForm();
+  const { state } = useLocation();
+
+  const loginHandler = async data => {
+    setLoading(true);
+    const response = await loginWithCreds(data.email, data.password);
+    if (response?.status === 200) {
+      setUserNotFoundMessage(null);
+      setAuthFaliureMessage(null);
+      setLoading(false);
+      navigate(state?.from ? state.from : "/profile", { replace: true });
+    } else if (response?.status === 401) {
+      setUserNotFoundMessage(null);
+      setAuthFaliureMessage(response.data.message);
+      setLoading(false);
+    } else if (response?.status === 403) {
```

Comment:
I tried logging in using wrong email and password, I'm getting a 404 status code but since it is not handled the loader is always loading. Handle other auth failures as well

Reply:
This was a great catch Sneha, I handled this error now. Thanks!